client/galaxy/scripts/mvc/workflow/workflow-terminals.js

```diff
@@ -531,20 +552,25 @@ var InputCollectionTerminal = BaseInputTerminal.extend({
                 // Otherwise we need to mapOver
             } else if (thisMapOver.isCollection) {
                 // In this case, mapOver already set and we didn't match skipping...
-                return false;
+                return new ConnectionAcceptable(false, COLLECTION_MAPPING_INCOMPATIBLE_OUTPUT);
             } else if (_.some(this.collectionTypes, collectionType => otherCollectionType.canMapOver(collectionType))) {
                 var effectiveMapOver = this._effectiveMapOver(other);
                 if (!effectiveMapOver.isCollection) {
-                    return false;
+                    return new ConnectionAcceptable(false, COLLECTION_MAPPING_INCOMPATIBLE_INPUT);
                 }
                 //  Need to check if this would break constraints...
                 var mappingConstraints = this._mappingConstraints();
                 if (mappingConstraints.every(effectiveMapOver.canMatch)) {
                     return this._producesAcceptableDatatype(other);
+                } else {
+                    return new ConnectionAcceptable(false, COLLECTION_MAPPING_INCOMPATIBLE_INPUT);
                 }
+            } else {
+                return new ConnectionAcceptable(false, "Incompatible collection type(s) for attachment.");
             }
+        } else {
+            return new ConnectionAcceptable(false, "Cannot attach a data output to a collection output.");
```

Comment:
Noticed in @mvdbeek's video:

```suggestion
            return new ConnectionAcceptable(false, "Cannot attach a data output to a collection input.");
```